#207 resolve core update issue missing labels

kcw-grunt opened the pull request:
## Problem
When updating the loafwallet-core module / library the address is failing a UnsafeUnmutablePointer when the opening the history of transactions.

As a result, the sent and receive addresses are showing as normal if they are `LNgqTxjWxmjsULeugZd7ee6wwVFsVGkz4js` **but** `ltc1qrledc2grd0lvluvcn388szkjhxllqmtyzr37rxn` or `M` or ltc1 addresses are not showing.

The issue is the standard of UnsafeUnmutablePointers has changed over the few years so the way to properly handle has changed.

## Positives

- The data is there in the database (SQLite) and the ltc1 addresses are present.
- The rest of the app is behaving normally


## Snapshot
|Sample History with Labels (DEBUG-UPDATE-ADDRESS)|
|------------|
|![Simulator Screenshot - iPhone 14 Pro - 2023-05-30 at 08 56 49](https://github.com/litecoin-foundation/litewallet-ios/assets/2899463/873fa4a1-a8c3-48d5-8051-32d819f9ea70)|

|Correct Addresses|
|---------------| 
|![test-addresses](https://github.com/litecoin-foundation/litewallet-ios/assets/2899463/f50cd2d6-9ef2-48df-a0b4-9e0b65ab0edf)|


## Goal

Fix the unwrapping so the addresses can be shown.  Most of my investigation is around the Transaction.swift class line 174

mosadialiou reviewed (approved):
LGTM

kcw-grunt committed:
move the git submodule to the latest

- v1.0.0 releease of loafwallet-core

kcw-grunt committed:
move the git submodule to the latest

- v1.0.0 releease of loafwallet-core

# Conflicts:
#	Modules/loafwallet-core

kcw-grunt committed:
ISSUE: Not able to type and display recv addresses

- There is a failiure in the  conversion from CChar to Int8
- Have send working
- updated loafwallet-core v1.1.0
- Added falsePositived interface
- Added enum and set values for
 - low privacy
 - semi-private
 - anonymous
- Updated the loafwallet-core
- adds the fpRate var

kcw-grunt committed:
Resolved PeerManager set with proper optional setting.

kcw-grunt committed:
WIP: Cannot unwrap ltc1 addresses to retain them in the UILabel

- ltc1 addresses are present but not unwrapping properly
@Iferencak this is the disappearing error you saw when we updated the loafwallet-core.
- Not able to figure this out yet.
- All ideas are welcome

kcw-grunt commented on loafwallet/src/ViewModels/Transaction.swift:
```diff
@@ -159,50 +159,38 @@ class Transaction {
 		case .moved:
 			guard let output = self.tx.outputs.filter({ output in
-				self.wallet.containsAddress(output.swiftAddress)
+				self.wallet.containsAddress(output.updatedSwiftAddress)
```
@losh11 here is where the problem happens

josikie reviewed (approved):
Cool! Thanks @kcw-grunt 

On this code:
``` 
addressText = String(format: transaction.direction.addressTextFormat, transaction.toAddress ?? "DEBUG-UPDATED-ADDRESS")
```
Why there are two questions mark?

kcw-grunt committed:
WIP:  Still debugging

- The issue is the BRCore methods compare the `ltc1` address against the set of addresses and it fails because the BRSet is not setup for the address.

The issue is this worked before and I need to investigate what happened before and how things changed

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
Bugfix!: Duplicated char in converter

Added optic enum

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
Move all the extensions from BRCore to separate files

- Moved these into separate files to help debug

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
Refactored the conversion

- Fixed a bug where a element is duplicated
`unichar(charArray.20),unichar(charArray.21), unichar(charArray.22), unichar(charArray.23), unichar(charArray.23), unichar(charArray.24)`

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>
